internal/manifests/collector/config_replace.go

```diff
@@ -58,6 +58,13 @@ func ReplaceConfig(otelcol v1beta1.OpenTelemetryCollector, targetAllocator *v1al
 		return "", validateCfgPromErr
 	}
 
+	// Set the interval - use CollectorTargetReloadInterval if specified, otherwise default to 30s
+	interval := "30s"
+	if targetAllocator.Spec.CollectorTargetReloadInterval != nil {
+		interval = targetAllocator.Spec.CollectorTargetReloadInterval.Duration.String()
+	}
```

Comment:
I'd rather not define this default in two places. Since it's already defined in the CRD, you can either assume it's non-nil here, or check and return an error.

Reply:
I also agree that having default values managed in multiple places is not ideal! So, I fix the logic assuming that the default values come from the CRD.